src/utils/rateLimiter.util.ts

```diff
@@ -0,0 +1,27 @@
+import { RedisService } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class RateLimiterService {
+  private readonly logger = new Logger(RateLimiterService.name);
+
+  constructor(private readonly redisService: RedisService) {}
+
+  async isRateLimited(senderId: string, repoId: string): Promise<boolean> {
+    const client = this.redisService.getClient();
+    const key = `rate_limit:${senderId}:${repoId}`;
+    const currentCount = await client.get(key);
+
+    if (currentCount && parseInt(currentCount) >= 5) {
+      return true;
+    }
+
+    await client
+      .multi()
+      .incr(key)
+      .expire(key, 60) // TTL of 60 seconds
+      .exec();
+
+    return false;
+  }
```

Comment:
PrimathonAICodeReviewer

The implementation looks good overall. One suggestion would be to add error handling for Redis operations to ensure robustness.

```diff
+     const currentCount = await client.get(key);
+
+     if (!currentCount) {
+         return false; // Handle Redis error or missing key
+     }
+
+     try {
+         const count = parseInt(currentCount);
+         if (count >= 5) {
+             return true;
+         }
+
+         await client
+             .multi()
+             .incr(key)
+             .expire(key, 60) // TTL of 60 seconds
+             .exec();
+     } catch (error) {
+         this.logger.error(`Error in Redis operation: ${error.message}`);
+         return false; // Handle Redis operation failure
+     }
```


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->